app/assets/javascripts/discourse/app/components/sidebar/more-section-link.hbs

```diff
@@ -1,15 +1,37 @@
-<Sidebar::SectionLink
-  @shouldDisplay={{@sectionLink.shouldDisplay}}
-  @linkName={{@sectionLink.name}}
-  @route={{@sectionLink.route}}
-  @href={{@sectionLink.href}}
-  @query={{@sectionLink.query}}
-  @title={{@sectionLink.title}}
-  @content={{@sectionLink.text}}
-  @currentWhen={{@sectionLink.currentWhen}}
-  @badgeText={{@sectionLink.badgeText}}
-  @model={{@sectionLink.model}}
-  @models={{@sectionLink.models}}
-  @prefixType={{@sectionLink.prefixType}}
-  @prefixValue={{@sectionLink.prefixValue}}
-/>
\ No newline at end of file
+{{#if @sectionLink.shouldDisplay}}
+  {{#if @sectionLink.external}}
+    <Sidebar::SectionLink
+      @linkName={{@sectionLink.name}}
+      @content={{replace-emoji @sectionLink.text}}
+      @prefixType="icon"
+      @prefixValue={{@sectionLink.prefixValue}}
+      @href={{@sectionLink.value}}
+    />
+  {{else if @sectionLink.action}}
+    <DButton
```

Comment:
We should probably extract this into a `<Sidebar::SectionButton>` component so that it can be reused in other places as well.

Reply:
Actually I'm wondering if this has to be a button? The `/new-topic` route exists and will already open a composer. Right now, we have this overarching assumption that all rows under a section are links are introducing a button complicates this assumption further.